tasks/build_tags.py

```diff
@@ -29,70 +32,180 @@
     "containerd",
     "cri",
     "netcgo",
     "systemd",
 ]
+
+# List of tags to always remove when building on Windows
+WINDOWS_EXCLUDE_TAGS = [
+    "linux_bpf"
+]
+
+# List of tags to always remove when building on Windows 32-bits
 WINDOWS_32BIT_EXCLUDE_TAGS = [
-    "orchestrator",
     "docker",
     "kubeapiserver",
     "kubelet",
+    "orchestrator",
 ]
 
-
-def get_default_build_tags(iot=False, process=False, arch="x64", android=False):
+def get_default_build_tags(build="agent", arch="x64"):
     """
-    Build the default list of tags based on the current platform.
+    Build the default list of tags based on the build type and current platform.
 
     The container integrations are currently only supported on Linux, disabling on
     the Windows and Darwin builds.
     """
-    include = ["all"]
-    if iot:
+    include = []
+    # Build setups
+    if build == "agent":
+        include = AGENT_TAGS
+    if build == "android":
+        include = ANDROID_TAGS
+    if build == "cluster-agent":
+        include = CLUSTER_AGENT_TAGS
+    if build == "cluster-agent-cloudfoundry":
+        include = CLUSTER_AGENT_CLOUDFOUNDRY_TAGS
+    if build == "dogstatsd":
+        include = DOGSTATSD_TAGS
+    if build == "iot":
         include = IOT_AGENT_TAGS
+    if build == "process-agent":
+        include = PROCESS_AGENT_TAGS
+    if build == "security-agent":
+        include = SECURITY_AGENT_TAGS
+    if build == "system-probe":
+        include = SYSTEM_PROBE_TAGS
+    if build == "trace-agent":
+        include = TRACE_AGENT_TAGS
+    # Test setups
+    if build == "test":
+        include = TEST_TAGS
+    if build == "test-with-process-tags":
+        include = TEST_TAGS + PROCESS_AGENT_TAGS
+
+    return filter_incompatible_tags(include, arch=arch)
+
+def filter_incompatible_tags(include, arch="x64"):
+    """
+    Filter out tags incompatible with the platform.
+    """
 
-    # android has its own set of tags
-    if android:
-        include = ANDROID_TAGS
+    exclude = []
+    if not sys.platform.startswith("linux"):
+        exclude += LINUX_ONLY_TAGS
 
-    exclude = [] if sys.platform.startswith("linux") else LINUX_ONLY_TAGS
-    # if not process agent, ignore process only tags
-    if not process:
-        exclude = exclude + PROCESS_ONLY_TAGS
+    if sys.platform == "win32":
+        exclude += WINDOWS_EXCLUDE_TAGS
 
-    # Force exclusion of Windows 32bits tag
     if sys.platform == "win32" and arch == "x86":
-        exclude = exclude + WINDOWS_32BIT_EXCLUDE_TAGS
+        exclude += WINDOWS_32BIT_EXCLUDE_TAGS
 
     return get_build_tags(include, exclude)
 
+
 def get_build_tags(include, exclude):
     """
     Build the list of tags based on inclusions and exclusions passed through
     the command line
     """
-    # special case, include == all
-    if "all" in include:
-        return list(ALL_TAGS - set(exclude))
 
     # filter out unrecognised tags
     include = ALL_TAGS.intersection(set(include))
```

Comment:
If this actually filters something we might want to print a warning (unrecognized tag: <bleh>). I can see someone who doesn't know about ALL_TAGS adding a tag and spending 1 hour debugging why it didn't take effect.